crypto/provider_core.c

```diff
@@ -278,6 +296,28 @@ void ossl_provider_free(OSSL_PROVIDER *prov)
     }
 }
 
+/* Setters */
+int ossl_provider_set_module_path(OSSL_PROVIDER *prov, const char *module_path)
+{
+    prov->path = OPENSSL_strdup(module_path);
```

Comment:
Check for allocation failure?  Otherwise it might as well be a void function.

Reply:
Also, if there was a previous value, it will be leaked

Reply:
Fixed